tests/Doctrine/Tests/ORM/Hydration/ScalarColumnHydratorTest.php

```diff
@@ -0,0 +1,138 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Tests\ORM\Hydration;
+
+use Doctrine\Tests\Models\CMS\CmsUser;
+use Doctrine\ORM\Query\ResultSetMapping;
+use Doctrine\Tests\Mocks\HydratorMockStatement;
+use Doctrine\ORM\Internal\Hydration\ScalarColumnHydrator;
+
+use function count;
+use function is_array;
+
+class ScalarColumnHydratorTest extends HydrationTestCase
+{
+    /**
+     * SELECT PARTIAL u.{id}
+     *   FROM Doctrine\Tests\Models\CMS\CmsUser u
+     */
+    public function testEmptyResultTest(): void
+    {
+        $rsm = new ResultSetMapping();
+
+        $rsm->addEntityResult(CmsUser::class, 'u');
+        $rsm->addFieldResult('u', 'u__id', 'id');
+
+        // Faked empty result set
+        $stmt = new HydratorMockStatement($resultSet = []);
+        $hydrator = new ScalarColumnHydrator($this->entityManager);
+        $result = $hydrator->hydrateAll($stmt, $rsm);
+
+        $this->assertEquals(0, count($result));
+        $this->assertTrue(is_array($result));
+    }
+
+    /**
+     * SELECT PARTIAL u.{id}
+     *   FROM Doctrine\Tests\Models\CMS\CmsUser u
+     */
+    public function testSingleColumnEntityQuery(): void
+    {
+        $rsm = new ResultSetMapping();
+
+        $rsm->addEntityResult(CmsUser::class, 'u');
+        $rsm->addFieldResult('u', 'u__id', 'id');
+
+        // Faked result set
+        $resultSet = [
+            0 => '1',
+            1 => '2',
+        ];
+
+        $stmt = new HydratorMockStatement($resultSet);
+        $hydrator = new ScalarColumnHydrator($this->entityManager);
+        $result = $hydrator->hydrateAll($stmt, $rsm);
+
+        $this->assertEquals(2, count($result));
+        $this->assertTrue(is_array($result));
+
+        $this->assertEquals(1, $result[0]);
+
+        $this->assertEquals(2, $result[1]);
+    }
+
+    /**
+     * SELECT PARTIAL u.{id}
+     *   FROM Doctrine\Tests\Models\CMS\CmsUser u
+     */
+    public function testSingleColumnEntityQueryWithScalarMap(): void
```

Comment:
> Not understanding with this test why It's giving nested results

<img width="1437" alt="Screenshot 2021-08-22 at 6 27 24 AM" src="https://user-images.githubusercontent.com/43483545/130338593-b99b1121-11dc-485c-b0b3-84013536176d.png">

Reply:
That's one more reason not to extend from `ArrayHydrator` IMO

Reply:
Extending `AbstractHydrator` also simplifying implementation using ScalarHydrator and `fetchOne`